app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsViewModel.kt

```diff
@@ -0,0 +1,111 @@
+package com.github.se.travelpouch.model.activity.map
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.PolyUtil
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import okhttp3.OkHttpClient
+
+// Define an alias type for a path, which is a list of LatLng points
+typealias Path = List<LatLng>
+
+// Data class to hold the different route between origin and destination
+data class RouteOptions(
+    val origin: LatLng,
+    val destination: LatLng,
+    var routes: List<Path> // List of routes, where each route is a list of LatLng points
```

Comment:
No, not really. the route list represents all the possible routes between a point A and a point B. for us at the moment this is not really useful because we will only take the first route but the Google API can give several routes and I thought it might be useful later.

For the moment what it does is that it fetches the list of routes for between 2 activities. You must therefore fetch as many times as there is activity. but after thinking over it,  it would be better to have a way to fetch a route between two activities with different stops. This could be an improvement for a next PR

Reply:
Yes, but the list of routes contains paths between two endpoints at each time no ?

Reply:
what do you mean by each time ?

Reply:
Ok, there was a misunderstading that was solved on Telegram

Reply:
It is good now